### PR TITLE
feat: send projectId and branchId to Segment

### PR DESCRIPTION
**Pull request by PlayLikeNeverB4**
Relates to https://github.com/neondatabase/cloud/issues/20941

Because the `cli_started` event is sent at the beginning we can't know the resource ids.

So I added a `cli_command_success` event after the command runs when we have the final context.

These will not be sent on error because I couldn't find a way to get the context on error.

### **Examples**

`neon branches list` will send `projectId`

`neon branches get someid` will send `projectId` and `branchId`

`neon branches get main` will send `projectId` and correct `branchId`

`branches restore A B` will send `projectId` and `branchId=A`

`neon me` will send both as `undefined`

### CODE REVIEW

**Comment by PlayLikeNeverB4** [comment]
Fixed it

**Review by duskpoet — COMMENTED** [review]
I am afraid that this approach is too unintuitive and new additions may fail to follow it. What about adding this logic to the helper method `branchIdFromProps`?

**Comment by PlayLikeNeverB4** [comment]
@duskpoet I implemented your suggestion. Also noticed that I needed to treat `restore` separately.

File: src/commands/branches.ts [hunk 1]
```diff
@@ -134,6 +138,7 @@ export const builder = (argv: yargs.Argv) =>
           .middleware((args: any) => {
             args.id = args.targetId;
             args.pointInTime = args['source@(timestamp'];
+            args.branchId = args.id; // for analytics
```
Comment (duskpoet) [inline src/commands/branches.ts:141]: why did you need a separate line here?

File: src/index.ts [hunk 2]
```diff
@@ -195,6 +201,11 @@ builder = builder
 void (async () => {
   try {
     const args = await builder.argv;
+    trackEvent('cli_command_success', {
```
Comment (duskpoet) [inline src/index.ts:204]: is it ok that failed commands won't be tracked?

**Review by duskpoet — APPROVED** [review]

File: src/commands/branches.ts (hunk 1 above)
Comment (PlayLikeNeverB4) [reply src/commands/branches.ts:141]: This command doesn't have an `id` parameter and doesn't use `branchIdFromProps` either, so we have to treat it separately.

File: src/index.ts (hunk 2 above)
Comment (PlayLikeNeverB4) [reply src/index.ts:204]: I tried, but in the `.fail()` callback I don't see the parameters that I set, they are all `undefined`. I tried a few other ways, nothing worked.

These calls will be tracked in the usual `cli_started` event I guess.

File: src/commands/branches.ts (hunk 1 above)
Comment (duskpoet) [reply src/commands/branches.ts:141]: oh, right. But `id` property will also be empty, no? you probably need the target-id

Comment (PlayLikeNeverB4) [reply src/commands/branches.ts:141]: It's set 2 lines above. I could use `targetId` if you think it's clearer.

File: src/index.ts (hunk 2 above)
Comment (PlayLikeNeverB4) [reply src/index.ts:204]: Tried that too. The `await builder.argv` line throws an exception which describes the failed request.

File: src/commands/branches.ts (hunk 1 above)
Comment (duskpoet) [reply src/commands/branches.ts:141]: ahhh, I finally got it! :D

File: src/index.ts (hunk 2 above)
Comment (duskpoet) [reply src/index.ts:204]: that is exaclty what we need, no?

Comment (PlayLikeNeverB4) [reply src/index.ts:204]: We need `projectId` and `branchId` and the `AxiosError` doesn't help with that.
